packages/components/src/menu-item/stories/index.story.tsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import type { Meta, StoryFn } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { link, more, check } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import MenuGroup from '../../menu-group';
+import MenuItem, { MenuItem as UnforwardedMenuItem } from '..';
+import Shortcut from '../../shortcut';
+
+const meta: Meta< typeof UnforwardedMenuItem > = {
+	component: UnforwardedMenuItem,
+	title: 'Components/MenuItem',
+	argTypes: {
+		children: { control: { type: null } },
+		icon: {
+			control: { type: 'select' },
+			options: [ 'check', 'link', 'more' ],
+			mapping: {
+				check,
+				link,
+				more,
+			},
+		},
+	},
+	parameters: {
+		controls: {
+			expanded: true,
+			include: [
+				'className',
+				'children',
+				'info',
+				'icon',
+				'iconPosition',
+				'isSelected',
+				'role',
+				'suffix',
+				'label',
+			],
+		},
+		docs: { source: { state: 'open' } },
+	},
+};
+export default meta;
+
+const Template: StoryFn< typeof MenuItem > = ( props ) => {
+	return (
+		<MenuGroup>
+			<MenuItem { ...props }>Menu Item 1</MenuItem>
+		</MenuGroup>
+	);
+};
+
+export const Default: StoryFn< typeof MenuItem > = Template.bind( {} );
+
+export const IsSelected = Template.bind( {} );
+IsSelected.args = {
+	...Default.args,
+	isSelected: true,
+};
```

Comment:
I noticed that we should improve the documentation for `isSelected`. It only takes effect when `role === 'menuitemcheckbox' || role === 'menuitemradio'`, but that isn't mentioned anywhere. Something like:

```suggestion
/**
 * When the `role` prop is either `"menuitemcheckbox"` or `"menuitemradio"`, the
 * `isSelected` prop should be used so screen readers can tell which item is currently selected.
 */
export const IsSelected = Template.bind( {} );
IsSelected.args = {
	...Default.args,
	isSelected: true,
	role: 'menuitemcheckbox',
};
```

We should mention this in the TypeScript description and readme as well.

Reply:
I updated in Stories, Readme and Type definitions, thanks! I've been able to dig deeper on the logic by reading https://github.com/WordPress/gutenberg/issues/11431